web/html/src/manager/virtualization/guests/properties/guest-disk.js

```diff
@@ -0,0 +1,312 @@
+            Number.parseInt(property.substring('disk'.length), 10), caps,
+            pools,
+            changeModel,
+            onlyHandledDisks))
+      }
+    </Panel>
+  );
+}
+
+function getRequestParams(model: Object, disk: string): Object {
+  const sourceMappers = {
+    file: (): Object => {
+      if (model[`${disk}_source_file`] !== undefined) {
+        const sourceFile = model[`${disk}_source_file`] !== '' ? model[`${disk}_source_file`] : null;
+        return { source_file: sourceFile };
+      }
+      return Object.assign(
+        {},
+        { pool: model[`${disk}_source_pool`] },
+        model[`${disk}_source_size`] !== '' ? { size: model[`${disk}_source_size`] } : {},
+        model[`${disk}_source_template`] !== '' ? { template: model[`${disk}_source_template`] } : {},
+      );
+    },
+  };
+  const sourceMapper = sourceMappers[model[`${disk}_type`]];
+  const source = sourceMapper !== undefined ? sourceMapper() : {};
+
+  return Object.assign({
+    device: model[`${disk}_device`],
+    type: model[`${disk}_type`],
+    bus: model[`${disk}_bus`],
+  },
+  source);
+}
+
+function getModelFromDefinition(definition: Object) {
+  return definition.disks != null
+    ? definition.disks.reduce(
+      (result, disk, index) => {
+        if (disk != null) {
+          const sourceToModel = {
+            file: (() => ({ [`disk${index}_source_file`]: disk.source ? disk.source.file : '' })),
+            dir: (() => ({ [`disk${index}_source_dir`]: disk.source ? disk.source.dir : '' })),
+            block: (() => ({ [`disk${index}_source_dev`]: disk.source ? disk.source.dev : '' })),
+            network: (
+              () => Object.assign(
+                {
+                  [`disk${index}_source_protocol`]: disk.source ? disk.source.protocol : '',
+                  [`disk${index}_source_name`]: disk.source ? disk.source.name : '',
+                },
+                disk.source && disk.source.host
+                  ? disk.source.host.reduce((resultHosts, host, hostIndex) => Object.assign(resultHosts,
+                    {
+                      [`disk${index}_source_host${hostIndex}_name`]: host.name,
+                      [`disk${index}_source_host${hostIndex}_port`]: host.port,
+                    }), { })
+                  : { },
+              )
+            ),
+            volume: (() => ({
+              [`disk${index}_source_pool`]: disk.source ? disk.source.pool : '',
+              [`disk${index}_source_volume`]: disk.source ? disk.source.volume : '',
+            })),
+          };
+
+          return Object.assign(result, {
+            [`disk${index}_type`]: disk.type,
+            [`disk${index}_device`]: disk.device,
+            [`disk${index}_target`]: disk.target,
+            [`disk${index}_bus`]: disk.bus,
+          }, sourceToModel[disk.type]());
+        }
+        return result;
+      },
+      { },
+    )
+    : { };
+}
+
+module.exports = {
+  guestDisksPanel,
```

Comment:
To keep consistent with the filename I would suggest either to change this component name to guestDisks or change the filename to guest-disks-panel.js ;)